rpc/lighthouse.go

```diff
@@ -149,12 +149,18 @@ func (lc *LighthouseClient) GetChainHead() (*types.ChainHead, error) {
 		return nil, fmt.Errorf("error parsing finality checkpoints of head: %v", err)
 	}
 
+	// The epoch in the Finalized Object is not the finalized epoch, but the epoch for the checkpoint - the 'real' finalized epoch is the one before
```

Comment:
Imo this comment is misleading. I would argue that 'real' finalization means that only an epoch's first slot is finalized. We mark an epoch finalized if all of its slots are finalized instead, which is not the 'real' definition of finalization but a more practical one.